apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/UnconfirmedTransaction.java

```
@@ -151,6 +166,17 @@ public Block getBlock() {
         return transaction.getBlock();
     }
 
+    public void setBlock(Block block) {
+        this.transaction.setBlock(block);
```

We cant set block in UnconfirmedTransaction, because it becomes confirmed (TransactionImpl)

---

Maybe write here `throw new UnsupportedOperationException()` or separate Unconfirmed and Confirmed transaction interfaces

---

Agree, it will better to do `throw new UnsupportedOperationException()` inside TransactionImpl if that's not possible here.

---

Done, pushed.